File: FLBP ANDROID/app/src/main/java/com/flbp/manager/suite/PublicRouteState.kt
package com.flbp.manager.suite

data class TournamentSelectionRef(
    val id: String,
    val isLive: Boolean
) {
    companion object {
        fun fromSaved(id: String, isLive: Boolean): TournamentSelectionRef? {
            val normalized = id.trim()
            return if (normalized.isEmpty()) null else TournamentSelectionRef(id = normalized, isLive = isLive)
        }
    }
}

data class PublicRouteState(
    val route: AppRoute = AppRoute.HOME,
    val selectedTournament: TournamentSelectionRef? = null
) {
    init {
        require(route.group == RouteGroup.PUBLIC_PRIMARY || route == AppRoute.TOURNAMENT_DETAIL) {
            "PublicRouteState accepts only public routes."
        }
    }

    val resolvedRoute: AppRoute
        get() = when {
            route == AppRoute.TOURNAMENT_DETAIL && selectedTournament == null -> AppRoute.TOURNAMENT
            else -> route
        }

    fun navigateToPrimary(route: AppRoute): PublicRouteState {
        require(route.group == RouteGroup.PUBLIC_PRIMARY) {
            "navigateToPrimary accepts only public primary routes."
        }
        return copy(route = route)
    }

    fun openTournamentDetailOrFallback(): PublicRouteState {
        return if (selectedTournament == null) {
            copy(route = AppRoute.TOURNAMENT)
        } else {
            copy(route = AppRoute.TOURNAMENT_DETAIL)
        }
    }

    fun withTournamentSelection(selection: TournamentSelectionRef?): PublicRouteState {
        val nextRoute = if (selection == null && route == AppRoute.TOURNAMENT_DETAIL) {
            AppRoute.TOURNAMENT
        } else {
            route
        }
        return copy(route = nextRoute, selectedTournament = selection)
    }
}

File: FLBP IOS/FLBPManagerSuite/PublicRouteState.swift
import Foundation

struct TournamentSelectionRef: Equatable {
    let id: String
    let isLive: Bool

    static func fromSaved(id: String, isLive: Bool) -> TournamentSelectionRef? {
        let normalized = id.trimmingCharacters(in: .whitespacesAndNewlines)
        return normalized.isEmpty ? nil : TournamentSelectionRef(id: normalized, isLive: isLive)
    }
}

struct PublicRouteState: Equatable {
    let route: NativeRoute
    let selectedTournament: TournamentSelectionRef?

    init(route: NativeRoute = .home, selectedTournament: TournamentSelectionRef? = nil) {
        precondition(route.group == .publicPrimary || route == .tournamentDetail, "PublicRouteState accepts only public routes.")
        self.route = route
        self.selectedTournament = selectedTournament
    }

    var resolvedRoute: NativeRoute {
        if route == .tournamentDetail && selectedTournament == nil {
            return .tournament
        }
        return route
    }

    func navigateToPrimary(_ route: NativeRoute) -> PublicRouteState {
        precondition(route.group == .publicPrimary, "navigateToPrimary accepts only public primary routes.")
        return PublicRouteState(route: route, selectedTournament: selectedTournament)
    }

    func openTournamentDetailOrFallback() -> PublicRouteState {
        if selectedTournament == nil {
            return PublicRouteState(route: .tournament, selectedTournament: nil)
        }
        return PublicRouteState(route: .tournamentDetail, selectedTournament: selectedTournament)
    }

    func withTournamentSelection(_ selection: TournamentSelectionRef?) -> PublicRouteState {
        let nextRoute: NativeRoute = (selection == nil && route == .tournamentDetail) ? .tournament : route
        return PublicRouteState(route: nextRoute, selectedTournament: selection)
    }
}
